src/System.Net.Requests/tests/HttpWebRequestTest.cs

```diff
@@ -19,14 +19,6 @@ public partial class HttpWebRequestTest
         private const string RequestBody = "This is data to POST.";
         private readonly byte[] _requestBodyBytes = Encoding.UTF8.GetBytes(RequestBody);
         private readonly NetworkCredential _explicitCredential = new NetworkCredential("user", "password", "domain");
-
-        private HttpWebRequest _savedHttpWebRequest = null;
-        private WebHeaderCollection _savedResponseHeaders = null;
-        private Exception _savedRequestStreamException = null;
-        private Exception _savedResponseException = null;
-
-        private int _requestStreamCallbackCallCount = 0;
-        private int _responseCallbackCallCount = 0;
```

Comment:
While I was fixing various hangs, etc., I made this change. After all the other fixes I made, it might be unnecessary.

You are right that Xunit creates a new test class instance per test method so these "instance" fields are not shared even though they are in the same test class. But a naive developer (like me) wouldn't figure that out right away. It is not intuitive.  It's better to use the state variables a best practice.